### ISSUE TITLE
Leaks private keys, Github API tokens and Docker registry credentials on host

### ISSUE THREAD

**Issue by atombender** [drone/drone#1720]
Running `ps` on the host machine shows that Drone passes the following things openly on the command line:

* A public/private keypair
* Name and credentials for private Docker registry
* Github API token (`netrc`)

One thing is that environment variables are not secure, but command lines! Here's the full output:

```shell
$ ps aufxww | grep docker
root     23683  0.0  0.1 200496  3300 ?        Sl   19:44   0:00  |   \_ docker-containerd-shim 81da217a220fd7113fca7a695f92262efd10fdb30cd1563babe49541ce646c7d /var/run/docker/libcontainerd/81da217a220fd7113fca7a695f92262efd10fdb30cd1563babe49541ce646c7d docker-runc
root     23697  0.0  0.1   4084  2564 ?        Ssl  19:44   0:00  |       \_ docker -- {"workspace":{"root":"/drone/src","path":"/drone/src/github.com/atombender/dolphin","netrc":{"machine":"github.com","login":"***","user":"x-oauth-basic"},"keys":{"public":"ssh-rsa ***","private":"-----BEGIN RSA PRIVATE KEY-----\n***\n-----END RSA PRIVATE KEY-----\n"}},"system":{"version":"","link_url":"https://***","plugins":null,"globals":null,"privileged_plugins":null},"repo":{"scm":"git","owner":"atombender","name":"dolphin","full_name":"atombender/dolphin","avatar_url":"https://avatars.githubusercontent.com/u/50314?v=3","link_url":"https://github.com/atombender/dolphin","clone_url":"https://github.com/atombender/dolphin.git","default_branch":"master","timeout":60,"private":true,"trusted":false,"allow_pr":true,"allow_push":true,"allow_deploys":false,"allow_tags":false},"build":{"number":22,"event":"push","status":"running","enqueued_at":1468784642,"created_at":1468784642,"started_at":1468784643,"finished_at":0,"deploy_to":"","commit":"a7e5ccb460d547cd6d6f88892a0056d83398ccde","branch":"drone","ref":"refs/heads/drone","refspec":"","remote":"https://github.com/atombender/dolphin.git","title":"","message":"Drone file test.","timestamp":0,"author":"atombender","author_avatar":"https://avatars.githubusercontent.com/u/50314?v=3","author_email":"alex@bengler.no","link_url":"https://github.com/atombender/dolphin/commit/a7e5ccb460d547cd6d6f88892a0056d83398ccde"},"job":{"id":22,"number":1,"status":"running","exit_code":0,"enqueued_at":1468784642,"started_at":1468784643,"finished_at":0,"environment":{}},"vargs":{"email":"***","file":"Dockerfile","insecure":false,"password":"***","registry":"***","repo":"dolphin","storage_driver":"aufs","tag":["latest"],"username":"***"}}
```

(Horrible secrets elided with `***`.)

Drone 0.4.

**Comment by tboerger** [comment]
This is how 12-factor apps work. What do you expect?

**Comment by tboerger** [comment]
Ah wait... This is the payload. This have been heavily changed for 0.5

**Comment by tboerger** [comment]
And as a third thing. You should respect issue templates. You have not discussed anything on Gitter, you clearly ignored the questions from the template, that's sad.

**Comment by atombender** [comment]
Does "heavily changed" mean that it's been fixed for 0.5?

I expect 12-factor apps not to leak anything. You exchange secret information through channels that are not open; not the process namespace, which is an open channel.

As for the template, sorry; but I did provide all the essential details, and I'm not on Gitter and have no wish to be, especially to discuss things that are blatant bugs. Nothing came up when I searched for pre-existing Github issues, which means that (unless there's an issue that I couldn't find) means that the behaviour has never been published before. Since 0.4 is the current version, it stands to reason that there should be an issue about so other people can see it and find it.
